lib/Expo.php

```diff
@@ -101,18 +101,23 @@ public function notify(array $interests, array $data, $debug = false)
             $postData[] = $data + ['to' => $token];
         }
 
-        $ch = $this->prepareCurl();
+        $postDataChunks = array_chunk($postData, 90);
+        $responses = [];
 
-        curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($postData));
+        foreach ($postDataChunks as $postDataChunk) {
+            $ch = $this->prepareCurl();
 
-        $response = $this->executeCurl($ch);
+            curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($postDataChunk));
 
-        // If the notification failed completely, throw an exception with the details
-        if ($debug && $this->failedCompletely($response, $recipients)) {
-            throw ExpoException::failedCompletelyException($response);
+            $response = $this->executeCurl($ch);
+
+            // If the notification failed completely, throw an exception with the details
+            if ($debug && $this->failedCompletely($response, $recipients)) {
```

Comment:
I would leave this check outside of the loop we don't want to interrupt the  request if one of them failed completely i want to get this feedback at the very end of the operation.

Reply:
That's reasonable 👍

Reply:
I also noticed a minor issue, the chunked $response is been checked against the complete $recipient array. This will always result in `this->failedCompletely($response, $recipients)` returning false.